### PR TITLE
chore(portal): Do not allow signing up from legally-restricted jurisdictions

### PR DESCRIPTION
**Pull request by AndrewDryga**
Related to #6807

### CODE REVIEW

**Review by jamilbk — COMMENTED** [review]
Need to cross check this list with legal and then make a formal announcement on blog and via email.

I believe GCP already blocks some of these.

What is the effect when the user hits this? Can we show an HTTP 451 Unavailable for Legal reasons? That will be more clear than a 403 Forbidden which might just look like a service issue.

**Comment by AndrewDryga** [comment]
@jamilbk with this PR merged if you go to `/sign_up` from one of blocked counties you will see HTTP 451. Everything else will keep working until we remove path filter.

**Comment by jamilbk** [comment]
> @jamilbk with this PR merged if you go to `/sign_up` from one of blocked counties you will see HTTP 451. Everything else will keep working until we remove path filter.

Got it. Waiting to hear back from legal then will do a review here.

File: terraform/modules/google-cloud/apps/elixir/network.tf [hunk 1]
```diff
@@ -65,6 +65,19 @@ resource "google_compute_security_policy" "default" {
...
 
+  rule {
+    description = "block sanctioned countries"
+
+    action   = "deny(451)"
+    priority = "101"
+
+    match {
+      expr {
+        expression = "request.path.matches(\"/sign_up\") && origin.region_code in ('RU', 'BY', 'KP', 'IR', 'SY')"
```
Comment (jamilbk) [inline terraform/modules/google-cloud/apps/elixir/network.tf]: We need these as well, and will need to block service access too (can do it in another PR after announcement is made):

```suggestion
        expression = "request.path.matches(\"/sign_up\") && origin.region_code in ('RU', 'BY', 'KP', 'IR', 'SY', 'CU', 'VE')"
```

Hmm do you know if regions are supported? Ideally we need to block these as well:

```
Crimea, Donetsk, Lubansk
```

Comment (AndrewDryga) [reply terraform/modules/google-cloud/apps/elixir/network.tf]: @jamilbk I double checked and they do have special region codes for them: XC and XD for NLB, will test if they work on App Armor too (and duplicate rules for LB if not): https://cloud.google.com/armor/docs/network-edge-policies

**Review by jamilbk — APPROVED** [review]
